### ISSUE TITLE
Deprecate getcurrent and isinst in twisted.python.reflect

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6859]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)| @multani reported|
|-|-|
|Trac ID|trac#6859|
|Type|task|
|Created|2013-11-30 17:23:28Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-functions-reflect-6859|

While porting ```twisted.python.reflect``` to Python 3 (see [#5929](https://github.com/twisted/twisted/issues/5929)), it appeared that ```getcurrent``` and ```isinst``` were using the non-existent in Python 3 ```types.ClassType```.

After closer inspection, these functions don't seem to be used in Twisted nor outside, and they both seems pretty useless with a "recent" version of Python : ```isinst``` is probably better replaced by the builtin ```isinstance``` (which it does already) and ```getcurrent``` seemed to be only useful used by ```isinst```

In [#5929](https://github.com/twisted/twisted/issues/5929)?replyto=30#comment:30, Itamar suggested to simply drop these functions from Python 3 and deprecate them.

The goal of this ticket is to deprecate these 2 functions in Python 2.x

Attachments:

* [deprecate-functions-reflect-6859-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/70b/70b8f10c7a966ab0aae081ce0f96ce04b405af23/5bd006bb4d38feb0f363c5e504f4f40781d5667c.patch) (2446 bytes) - added by multani on 2014-02-26 09:55:38Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6859 6859
type__task task
reporter__multani multani
priority__normal normal
milestone__ 
branch__branches_deprecate_functions_reflect_6859 branches/deprecate-functions-reflect-6859
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1385832208000000 1385832208000000
changetime__1394982287000000 1394982287000000
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)<a name="note_1"></a>|@multani set owner to @multani|
|-|-|

For reference, I'll wait for [[#5929](https://github.com/twisted/twisted/issues/5929)](https://github.com/twisted/twisted/issues/5929) to get merged first and I will then propose a patch to fix this ticket (see #5929#comment:35 )

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj commented|
|-|-|

(In [41754]) Merge python3-reflect-5929-2

Author: multani
Reviewers: radix, thijs, rwall, glyph
Fixes: [#5929](https://github.com/twisted/twisted/issues/5929)
Refs: [#6239](https://github.com/twisted/twisted/issues/6239), [[#6986](https://github.com/twisted/twisted/issues/6986)](https://github.com/twisted/twisted/issues/6986), [[#6859](https://github.com/twisted/twisted/issues/6859)](https://github.com/twisted/twisted/issues/6859)

Port most of twisted.python.reflect to Python3.

Porting objgrep posed some problems and has been moved to a separate ticket #6986.

Porting getcurrent and isinst also posed problems and it was decided to deprecate them in  #6859.

**Comment by twisted-trac** [comment]
|[<img alt="multani's avatar" src="https://avatars.githubusercontent.com/u/65311?s=50" width="50" height="50">](https://github.com/multani)<a name="note_3"></a>|@multani commented|
|-|-|

I didn't set a replacement for `getcurrent` and I don't really know if the topfile and the `deprecated` call are OK for this, but that should be sufficient, I think.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

(In [41905]) Branching to deprecate-functions-reflect-6859.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set owner to @hawkowl|
|-|-|

Looks good to me, buildbots are green where it matters. Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [41908]) Merge deprecate-functions-reflect-6859: Deprecate getcurrent and isinst in twisted.python.reflect

Author: multani
Reviewer: hawkowl
Fixes: [#6859](https://github.com/twisted/twisted/issues/6859)
